src/wallet/wallet.cpp

```diff
@@ -364,8 +365,34 @@ void CWallet::Flush(bool shutdown)
     bitdb.Flush(shutdown);
 }
 
-bool CWallet::Verify(const string& walletFile, string& warningString, string& errorString)
+bool static UIError(const std::string &str)
 {
+    uiInterface.ThreadSafeMessageBox(str, "", CClientUIInterface::MSG_ERROR);
+    return false;
+}
+
+bool static UIWarning(const std::string &str)
+{
+    uiInterface.ThreadSafeMessageBox(str, "", CClientUIInterface::MSG_WARNING);
+    return true;
+}
+
```

Comment:
I have thought about that and if we want to decouple the wallet, we probably need to accept short term duplications of some code. In that case, the factored out shared code is `uiInterface.ThreadSafeMessageBox`. A duplicated entry point only used by CWallet should be okay.

But I agree with @mrbandrews: the return code should be switched to `void` (for `UIWarning`, keep it for `UIError` for now).
Will fix that.